components/keys/src/time.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::fmt;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct TimeStamp(u64);
+
+const TSO_PHYSICAL_SHIFT_BITS: u64 = 18;
+
+impl TimeStamp {
+    /// Create a time stamp from physical and logical components.
+    pub fn compose(physical: u64, logical: u64) -> TimeStamp {
+        TimeStamp((physical << TSO_PHYSICAL_SHIFT_BITS) + logical)
+    }
+
+    pub const fn min() -> TimeStamp {
+        TimeStamp(0)
+    }
+
+    pub const fn max() -> TimeStamp {
+        TimeStamp(std::u64::MAX)
+    }
+
+    pub const fn new(ts: u64) -> TimeStamp {
+        TimeStamp(ts)
+    }
+
+    /// Extracts physical part of a timestamp, in milliseconds.
+    pub fn extract_physical(self) -> u64 {
```

Comment:
As a method of `TimeStamp`, maybe the word `extract` is not necessary any more. Just `physical` or `get_physical` is ok.

Reply:
:+1: